### PR TITLE
Update RL examples: Drive and VehicleFollowing

### PR DESCRIPTION
**Pull request by Adaickalavan**
Updated RL examples: Drive and VehicleFollowing
+ RGB image fed into the neural network uses higher resolution and is cropped to focus on frontal view.
+ If mission goal is present, the goal position is superimposed onto rgb image before being fed into the neural network.
+ Refactored the RL training script.
+ Training scenarios for Driving SMARTS 2023.1, 2023.2, and 2023.2 benchmarks are updated in the docs, in the benchmarks, and in the training examples.

### CODE REVIEW

File: examples/rl/platoon/train/run.py [hunk 1]
```diff
@@ -100,79 +120,95 @@ def main(args: argparse.Namespace):
...
-        print("\nStart training.\n")
-        scenarios_iter = cycle(config.scenarios)
-        model = sb3lib.PPO(
-            env=envs_train[next(scenarios_iter)],
-            tensorboard_log=config.logdir / "tensorboard",
-            verbose=1,
-            **network.combined_extractor(config),
-        )
+    for index in range(config.epochs):
+        scen = next(scenarios_iter)
```
Comment (Gamenot) [inline examples/rl/platoon/train/run.py:140]: I found recently that it is possible to do the following
```suggestion
    scenarios_iter = islice(cycle(config.scenarios), config.epocs)
    for index, scen in enumerate(scenarios_iter):
```

**Review by Gamenot — APPROVED** [review]
Good work as always.

**Review by saulfield — APPROVED** [review]

File: examples/rl/platoon/train/reward.py [hunk 2]
```diff
@@ -78,32 +81,38 @@ def _reward(self, obs, env_reward):
...
+            # Penalty for driving on wrong way
+            if obs[agent_id]["events"]["wrong_way"]:
+                reward[agent_id] -= np.float64(10)
+                print(f"{agent_id}: Went wrong way.")
+                continue
 
-            # Reward for distance travelled by driving
-            reward[agent_id] += np.float64(env_reward[agent_id])
+            # Reward for distance travelled in one step
+            # reward[agent_id] += np.float64(env_reward[agent_id])
```
[the remark this answers is not in the document]
Comment (Adaickalavan) [reply examples/rl/platoon/train/reward.py:99]: I had mixed results when rewarding the agent for driving in a vehicle following task. I have removed it.

File: examples/rl/platoon/inference/contrib_policy/filter_obs.py [hunk 3]
```diff
@@ -75,6 +91,28 @@ def filter(self, obs: Dict[str, Any]) -> Dict[str, Any]:
                 if all(rgb_ego[img_y, img_x, :] == self._no_color):
                     rgb_ego[img_y, img_x, :] = self._wps_color
 
+        # Superimpose goal position onto rgb image
```
[the remark this answers is not in the document]
Comment (Adaickalavan) [reply examples/rl/platoon/inference/contrib_policy/filter_obs.py:94]: Adding the goal position into Envision will make our simulator more visually attractive.
